app/controllers/application_controller.rb

```diff
@@ -1,7 +1,15 @@
 class ApplicationController < ActionController::Base
-  before_action :identify_user
+ before_action :identify_user, :current_user
 
   def identify_user
-    @user = cookies[:user_id] ||= SecureRandom.hex(10)
+    unless current_user.present?
+      @user = User.create
+      session[:user_id] = @user.id
+    end
+  end
+
+  def current_user
+    return unless session[:user_id]
+    @current_user ||= User.find(session[:user_id])
```

Comment:
This needs clean up - 2 methods with similar purpose, no obvious control flow, 2 instance variables with the same (?) content...

Simplify! All we need in the end is that we got a user stored as an instance variable with reference in the session